modules/st2-criteria/directive.js

```diff
@@ -8,30 +8,93 @@ angular.module('main')
       require: 'ngModel',
       scope: {
         ngModel: '=',
-        disabled: '='
+        disabled: '=',
+        trigger: '='
       },
       templateUrl: 'modules/st2-criteria/template.html',
-      link: function postLink(scope) {
+      link: function postLink(scope, element, attrs, ctrl) {
         scope.ngModel = scope.ngModel || {};
 
-        scope.remove = function (key) {
-          delete scope.ngModel[key];
+        scope.typeSpec = {
+          enum: {
+            'matchregex': 'Matches Regex',
```

Comment:
Nice. Eventually we could also add a "meta" API for retrieving available criteria comparision operators.